### PR TITLE
Add Twitter OIDC provider configuration

### LINKED ISSUE #14093: Skip OIDC 'openid' scope redirect parameter
**Description**
When trying to create an application to connect to spotify oauth2 server their server answers :

Illegal scope

Looking at the URL generated by the quarkus extension, the parameter is passed as "&scope=openid+user-read-private+user-read-email"

This "OpenID" value is passed directly in the code, as seen in the quarkus openid extension :

https://github.com/quarkusio/quarkus/blob/068b19c036ac385c941708aa073f6e3e31d9f3ef/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java#L211

When removing the "openid" parameter, my application works, so it would be interesting to have an option for don't send it. 

**Implementation ideas**

As suggested for @sberyozkin, put a boolen property if it should be suppressed. 

https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/oidc/near/221510904

### LINKED ISSUE #23593: Add OIDC Twitter provider configuration
### Description

Now that PKCE is supported it should be possible to add a `twitter` configuration, Steph, please verify and add it once it works for you, thanks

https://developer.twitter.com/en/docs/authentication/oauth-2-0/authorization-code

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by FroMage**
Still missing: 
- docs (I didn't find where the current list is documented in our guides, did we forget documenting it?)
- tests

Fixes #23593
Fixes #14093

### CODE REVIEW

**Comment by sberyozkin** [comment]
Hey @FroMage, thanks

Re `docs`, I think it would be great for the related [Renarde docs](https://quarkiverse.github.io/quarkiverse-docs/quarkus-renarde/dev/security.html#_using_oidc_for_login) adapted for Quarkus as well, as discussed at #21521 (assigned to you), perhaps you can have a look at it next ? ( we can also check then if something has to be optimized in Quarkus)

Re the tests - just copy and paste a pair of unit tests in `OidcUtilsTest` (`github` or `facebook` related and have 2 new ones for `twitter`), that should be enough

File: extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcTenantConfig.java [hunk 1]
```diff
@@ -570,6 +570,13 @@ public void setSource(Source source) {
         @ConfigItem
         public Optional<List<String>> scopes = Optional.empty();
 
+        /**
+         * Add the 'openid' scope automatically to the list of scopes. This is useful for every OAuth provider
```
Comment (sberyozkin) [inline extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcTenantConfig.java:574]:
```suggestion
         * Add the 'openid' scope automatically to the list of scopes. This is required for OpenId Connect providers
```

File: extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcTenantConfig.java [hunk 2]
```diff
@@ -570,6 +570,13 @@ public void setSource(Source source) {
...
+         * except Twitter OAuth, for example, which does not accept that scope and throws an error.
```
Comment (sberyozkin) [inline extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcTenantConfig.java:575]:
```suggestion
         * but may not work for OAuth2 providers such as Twitter OAuth2 which does not accept the 'oidc' scope and throws an error.
```

**Comment by sberyozkin** [comment]
@FroMage LGTM thanks, I've suggested a minor clarification to the new configuration property; please add 2 tests to `OidcUtilsTest` and it will be ready to go

**Comment by sberyozkin** [comment]
May be worth rebasing as there is a message `This branch is out of date`

**Comment by gsmet** [comment]
> May be worth rebasing as there is a message This branch is out of date

No need for rebasing if the branch is not too old and there are no conflicts. You can do it when you push new fixes but let's not rebase just for the sake of it (see my email on quarkus-dev about this rebase feature and CI resources).

In this case, you will push new things so you can rebase when doing it. Talking about the general case here.

**Comment by FroMage** [comment]
OK, done.

File: extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java [hunk 3]
```diff
@@ -350,9 +350,10 @@ private boolean shouldAutoRedirect(TenantConfigContext configContext, RoutingCon
                                 ? configContext.oidcConfig.getAuthentication().scopes.get()
                                 : Collections.emptyList();
                         List<String> scopes = new ArrayList<>(oidcConfigScopes.size() + 1);
-                        scopes.add("openid");
+                        if (configContext.oidcConfig.getAuthentication().addOpenidScope.orElse(true)) {
+                            scopes.add("openid");
+                        }
                         scopes.addAll(oidcConfigScopes);
-                        configContext.oidcConfig.getAuthentication().scopes.ifPresent(scopes::addAll);
```
Comment (sberyozkin) [inline extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java:355]: @FroMage I'm not sure why, as it does look like it duplicates the previous line, but apparently removing it cause a test failure, can you check it please

Comment (sberyozkin) [reply extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java:355]: Hmm, passed for me

Comment (sberyozkin) [reply extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java:355]: @FroMage Your code is perfectly fine, the test was broken a bit (well, I did not realize till now the scopes were duplicated - this should never a problem but is obviously redundant and can indeed cause some problems, thanks)

**Review by sberyozkin — APPROVED** [review]
Thanks @FroMage for spending the time on testing it with Twitter

**Pull request merged**